### PR TITLE
7340 : Improve navigation after being logged in with saml sso

### PR DESCRIPTION
**Pull request by wbenabdessalem**
Bug : After entering authentification via SAML (microsoft or Google) the navigation is done in the pop-up as an external url.

### CODE REVIEW

File: eXo/Sources/Controllers/HomePage/HomePageViewController.swift [hunk 1]
```diff
@@ -36,7 +36,7 @@ class HomePageViewController: eXoWebBaseController, WKNavigationDelegate, WKUIDe
         self.setupWebView(self.webViewContainer)
         webView?.navigationDelegate = self
         webView?.uiDelegate = self
-        loadingIndicator.startAnimating()
+     //   loadingIndicator.startAnimating()
```
Comment (ahamdi) [inline eXo/Sources/Controllers/HomePage/HomePageViewController.swift:39]: To remove this code if it is no more needed

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: eXo/Sources/Controllers/HomePage/HomePageViewController.swift [hunk 2]
```diff
@@ -214,28 +217,6 @@ class HomePageViewController: eXoWebBaseController, WKNavigationDelegate, WKUIDe
             UserDefaults.standard.setValue(request.url?.absoluteString, forKey: "serverURL")
             UserDefaults.standard.setValue(true, forKey: "wasConnectedBefore")
         }
-        /*
-        Open request for external link (asked by user not automatic request for external link) in a new windows (Preview Controller)
-        - WKNavigationType of a automatic request is always = .Others
-        */
-        if (request.url?.absoluteString.range(of: serverDomain!) == nil && navigationAction.navigationType != WKNavigationType.other) {
```
Comment (ahamdi) [inline eXo/Sources/Controllers/HomePage/HomePageViewController.swift:221]: This code permits to open external links in a new window. I think we will have to keep it unless requested by our PO.
This code also fixes the problem of SAML, could you make sure that the SAML fix remains ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: eXo/Sources/Controllers/HomePage/HomePageViewController.swift (hunk 2 above)
Comment (wbenabdessalem) [reply eXo/Sources/Controllers/HomePage/HomePageViewController.swift:221]: Yes, it's okay

Comment (ahamdi) [reply eXo/Sources/Controllers/HomePage/HomePageViewController.swift:221]: It seems this check does not take care of SAML requests , isn't it ?

**Review by ahamdi — APPROVED** [review]

File: eXo/Sources/Controllers/HomePage/HomePageViewController.swift (hunk 2 above)
Comment (wbenabdessalem) [reply eXo/Sources/Controllers/HomePage/HomePageViewController.swift:221]: No it's already handled, in the same webview , just the call to the external link was fixed here.

**Pull request merged**